### ISSUE TITLE
Rational for single connection

### ISSUE THREAD

**Issue by Tibo-lg** [discreetlogcontracts/dlcspecs#69]
The latest transaction specifications reuse a paragraph from BOLT 1 (https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#connection-handling-and-multiplexing) about using a single connection. While this is not of very high importance at the moment, it would be nice to know the rationale for this and how it is actually implemented.

**Comment by ariard** [comment]
I would guess a) performance don't allocate twice per-peer b) avoid an easy DoS vector ?

I agree it would be better to allow multiple connections at least for robustness.

**Comment by LLFourn** [comment]
In my limited experience with this kind of thing it's usually necessary to allow the other peer to make a connection to you even when you are in the process of opening a connection to them to avoid error prone decision making around which connection you both should use. Refusing new connections if you already have a working one might seem like a good idea but I guess that necessitates keeping track of which connections are really still open (e.g. ping/pong).

**Comment by Tibo-lg** [comment]
I agree with you @LLFourn that's also the experience I have. In a previous project I worked on, we were just allowing having two connections to a single peer (one initiated by each peer) and then just use whichever. I'm also not sure how you would decide which one to keep or not.

> Refusing new connections if you already have a working one might seem like a good idea but I guess that necessitates keeping track of which connections are really still open (e.g. ping/pong).

I think it can also make things complicated, I would say just replacing connections (and closing previous one) might be easier.

Anyway it's quite a detail but I was curious (and kind of still am) why this is in the Lightning spec and how they actually implement it.
